code/modules/food_and_drink/snacks.dm

```diff
@@ -208,7 +208,36 @@
+	var/static/list/drinkwhitelist = list(/obj/item/reagent_containers/food/drinks/energyshake,/obj/item/reagent_containers/food/drinks/bottle/red,/obj/item/reagent_containers/food/drinks/bottle/blue,/obj/item/reagent_containers/food/drinks/bottle/pink,/obj/item/reagent_containers/food/drinks/bottle/lime,/obj/item/reagent_containers/food/drinks/bottle/spooky,/obj/item/reagent_containers/food/drinks/bottle/spooky2,/obj/item/reagent_containers/food/drinks/bottle/xmas,/obj/item/reagent_containers/food/drinks/bottle/bottledwater,/obj/item/reagent_containers/food/drinks/bottle/softsoft_pizza,/obj/item/reagent_containers/food/drinks/bottle/grones,/obj/item/reagent_containers/food/drinks/bottle/orange,/obj/item/reagent_containers/food/drinks/bottle/gingerale,/obj/item/reagent_containers/food/drinks/bottle/drowsy,/obj/item/reagent_containers/food/drinks/water,/obj/item/reagent_containers/food/drinks/tea,/obj/item/reagent_containers/food/drinks/coffee,/obj/item/reagent_containers/food/drinks/eggnog,/obj/item/reagent_containers/food/drinks/chickensoup,/obj/item/reagent_containers/food/drinks/weightloss_shake,/obj/item/reagent_containers/food/drinks/cola,/obj/item/reagent_containers/food/drinks/cola/random,/obj/item/reagent_containers/food/drinks/peach,/obj/item/reagent_containers/food/drinks/milk,/obj/item/reagent_containers/food/drinks/milk/rancid,/obj/item/reagent_containers/food/drinks/milk/soy,/obj/item/reagent_containers/food/drinks/covfefe,/obj/item/reagent_containers/food/drinks/bottle/beer,/obj/item/reagent_containers/food/drinks/bottle/fancy_beer,/obj/item/reagent_containers/food/drinks/bottle/wine,/obj/item/reagent_containers/food/drinks/bottle/hobo_wine,/obj/item/reagent_containers/food/drinks/bottle/champagne,/obj/item/reagent_containers/food/drinks/bottle/cider,/obj/item/reagent_containers/food/drinks/bottle/rum,/obj/item/reagent_containers/food/drinks/rum_spaced,/obj/item/reagent_containers/food/drinks/bottle/mead,/obj/item/reagent_containers/food/drinks/bottle/vintage,/obj/item/reagent_containers/food/drinks/bottle/vodka,/obj/item/reagent_containers/food/drinks/bottle/tequila,/obj/item/reagent_containers/food/drinks/bottle/gin,/obj/item/reagent_containers/food/drinks/bottle/ntbrew,/obj/item/reagent_containers/food/drinks/bottle/thegoodstuff,/obj/item/reagent_containers/food/drinks/curacao,)
+	proc/filter_food(food_type)
+		for (var/food_filter in src.blacklist)
+			if (ispath(food_type,food_filter))
+				return FALSE
+			.= TRUE
 
+	New()
+		..()
+		var/obj/item/reagent_containers/food/snacks/ingredient/pizza3/pizzabase = new /obj/item/reagent_containers/food/snacks/ingredient/pizza3
+		var/num_ingredients = rand(1,3)
+		var/snack_type
+		var/ingredient_quality = 0.6
+		for (var/i = 1, i <= num_ingredients, i++)
+			if (prob(80))
+				snack_type =  pick(filtered_concrete_typesof(/obj/item/reagent_containers/food/snacks, .proc/filter_food))
```

Comment:
Covered by @geneva-breaker's comment in a different area, but heck no to every possible type of snack minus a few blacklisted ones being included here.

Reply:
I've got a discussion thread up, I'm targeting some level of randomness with premade whitelists pending community input, my local changes have what I discuss in the thread.